OpenRA.Platforms.Default/Sdl2PlatformWindow.cs

```diff
@@ -450,5 +453,22 @@ public void SetScaleModifier(float scale)
 			scaleModifier = scale;
 			OnWindowScaleChanged(windowScale, windowScale * oldScaleModifier, windowScale, windowScale * scaleModifier);
 		}
+
+		public IReadOnlyDictionary<string, int> GetDisplays()
```

Comment:
I would define this as
```csharp
public string[] GetDisplayNames()
{
	return Exts.MakeArray(SDL.SDL_GetNumVideoDisplays(), i => SDL.SDL_GetDisplayName(i));
}
```
(and change the interface/Renderer method names to match).

The key changes being:
* Changing the type as mentioned above
* Returning the unmodified names (the number is a UI display detail that should be handled by the SettingsLogic)
* Much simpler code

Reply:
My concern with using `string[]` is that now we are using the arrays index to keep track on which index the monitor is on. It will make the code a little more brittle. For example if someone would change the order of how the items in the dropdown a presented by changing the order of the list it would introduce a bug. Anyway don't think it will be any problems here tho so nothing that stop this from proceed.